htlcswitch/interfaces.go

```diff
@@ -150,6 +153,12 @@ type ChannelLink interface {
 	// the original funding output can be found.
 	ShortChanID() lnwire.ShortChannelID
 
+	// OtherShortChanID returns an alternative ShortChannelID for the
+	// channel. For option_scid_alias zero-conf channels, this will be the
+	// confirmed ShortChannelID. For regular option_scid_alias channels,
```

Comment:
Ah ok this mapping makes more sense now:
  * pure zero conf chans: the alias is set as the actual scid, and the confirmed is the "other" 
  * non-zero conf, but scid alias: scid is the actual, other is the alias 

I think we can modify the variable naming and methods here to make this a bit clearer. Otherwise, sections that we're using the scid (but it's actually the alias) can be somewhat confusing at the call site.